lib/customer.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'takeaway'
+
+class Customer
+
+  attr_accessor :takeaway
+
+  def initialize
+    @takeaway = yield
```

Comment:
Good attempt to use dependency injection here - but as you have an `attr_accessor` for `takeaway`, it is unnecessary.  If you do want to specify the `takeaway` when `Customer` is created, you don't need to use a block here, just pass `takeaway` as an argument to `initialize`